File: code/timeboard.py
from datadog import initialize, api

options = {
    'api_key': '<api-key>'
    'app_key': '<app-key>'
}

initialize(**options)

title = "My Timeboard"
description = "An informative timeboard123."
graphs = [{
    "definition": {
        "events": [],
        "requests": [
            {"q": "avg:my_metric{*} by {host}"}
        ],
        "viz": "timeseries"
    },
    "title": "Random Metric"
},
{
    "definition": {
        "events": [],
        "requests": [
            {"q": "anomalies(avg:mysql.performance.user_time{*}, 'basic', 3)"}
        ],
        "viz": "timeseries"
    },
    "title": "mySQL view"

},
{
    "definition": {
        "events": [],
        "viz": "query_value",
        "requests": [
            {"q": "sum:my_metric{host:i-03cd49e6b870a06d7}","aggregator": "sum"}
        ]
     },
     "title":"sum of custom graph"
}
]

template_variables = [{
    "name": "host123",
    "prefix": "host123",
    "default": "host:i-03cd49e6b870a06d7"
}]

read_only = True
response = api.Timeboard.create(title=title,
                     description=description,
                     graphs=graphs,
                     template_variables=template_variables,
                     read_only=read_only)

print(response)

File: code/timeboard.rb
require 'byebug' # Import necessary entities
require 'dogapi'

# Find your keys here:
# https://app.datadoghq.com/account/settings#api

api_key = '719d714d7132af72ce6e1f2d8b67b618' 
app_key = 'ad4336c11c87bab29da8da02b46921f83314c922' 

# To bark really loud, you need an impressive dog!
chihuahua = Dogapi::Client.new(api_key, app_key) 

# In the next section we are going to create a simple Timeboard.

# Title displayed on the top of the Timeboard
title = 'Logfathers Timemachine'

# Let everyone know your intentions as to what this timeboard was created for.
description = 'This timeboard presents everything that is necessary to become a successful sales engineer for Datadog. Arf.' 

# Define the new graph to be displayed
graphs = [{
    "definition" => {
        "events" => [],
        "requests" => [
          {"q" => "avg:logfather_cpu{*}"},
        ],
        "viz" => "timeseries"
    },
    "title" => "CPU usage scoped over the Logfather host"
}]
template_variables = [{
    "name" => "Logfathers Timemachine",
    "prefix" => "Logfathers Timemachine",
    "default" => "host:Logfathers Timemachine"
}]

chihuahua.create_dashboard(title, description, graphs, template_variables)